### PR TITLE
edit docker entry bash file to set threads accordingly

### PR DESCRIPTION
**Pull request by zxy844288792**
*Description of changes:*
set OMP_NUM_THREADS to num of cores

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: container/dockerd-entrypoint.sh [hunk 1]
```diff
@@ -2,14 +2,26 @@
...
+    shift 1
     cp -v /opt/ml/model/* /home/model-server/model
     cp -v /home/model-server/neo_template.py /home/model-server/model
     model-archiver --handler neo_template:predict --model-name neomodel --model-path /home/model-server/model -f --export-path /home/model-server
     mv /home/model-server/neomodel.mar /home/model-server/model
 
-    shift 1
-    mxnet-model-server --start --mms-config config.properties
+    if [ ! -z "${MMS_NUM_WORKER}" ]
+    then
```
Comment (apivovarov) [inline container/dockerd-entrypoint.sh]: line 8 has `if` and `then` on the same line. Can you do the same on line 15?

Comment (zxy844288792) [reply container/dockerd-entrypoint.sh]: sure

File: Jenkinsfile [hunk 2]
```diff
@@ -125,16 +121,14 @@ def CloudInstallAndTest(cloudTarget) {
...
+      unstash 'dlr_gpu_whl'
+    } else {
+      unstash 'dlr_cpu_whl'
     }
     sh """
-    ls -lh *.whl
-    pip3 install *.whl
+    ls -lh python/dist/*.whl
+    pip3 install python/dist/*.whl
     pip3 install tensorflow
```
Comment (yongwww) [inline Jenkinsfile]: the command to install tf for gpu is `pip3 install tensorflow_gpu`

File: container/dockerd-entrypoint.sh [hunk 3]
```diff
@@ -2,14 +2,25 @@
 set -e
 set -x
 
+export IFS=' '
+MMS_NUM_WORKER=1
```
Comment (yongwww) [inline container/dockerd-entrypoint.sh]: is MMS_NUM_WORKER configurable? one worker for one instance as default?

File: Jenkinsfile [hunk 4]
```diff
@@ -125,18 +121,24 @@ def CloudInstallAndTest(cloudTarget) {
   def nodeReq = "ubuntu && amd64 && ${cloudTarget}"
   node(nodeReq) {
     echo "Installing DLR package for ${cloudTarget} target"
-    withAWS(credentials:'Neo-AI-CI-Fleet') {
-      files = s3FindFiles(bucket: 'neo-ai-dlr-jenkins-artifacts', glob: "${env.JOB_NAME}/${env.BUILD_ID}/artifacts/dlr-*-manylinux1_x86_64.whl")
-      assert files.size() == 1
-      files.each {
-        s3Download file: it.name, bucket: 'neo-ai-dlr-jenkins-artifacts', path: it.path, force: true
-      }
+    if (cloudTarget == "p2" || cloudTarget == "p3") {
```
Comment (apivovarov) [inline Jenkinsfile]:
```
>>> "p2" in ("p2", "p3")
True
```

Comment (hcho3) [reply Jenkinsfile]: Is this Groovy?

File: Jenkinsfile [hunk 5]
```diff
@@ -125,18 +121,24 @@ def CloudInstallAndTest(cloudTarget) {
...
+    ls -lh python/dist/*.whl
+    pip3 install python/dist/*.whl
     """
+    if (cloudTarget == "p2" || cloudTarget == "p3") {
+      sh """
+      pip3 install tensorflow_gpu
+      """
+    } else {
+      sh """
+      pip3 install tensorflow
```
Comment (apivovarov) [inline Jenkinsfile]: Just merged PR #55 where I changed the command to
```sudo pip3 install --upgrade --force-reinstall tensorflow```

**Comment by yongwww** [comment]
@apivovarov  for tf installation, I remember you have specific tf binary/wheel for some instances, pls suggest or submit a follow-up pr after this.

File: container/dockerd-entrypoint.sh (hunk 3 above)
Comment (zxy844288792) [reply container/dockerd-entrypoint.sh]: Yes it is, but right now we are keeping it for 1 for best latency. (I replied in the morning but I could not find my reply)

File: Jenkinsfile (hunk 2 above)
Comment (zxy844288792) [reply Jenkinsfile]: change to tensorflow_gpu for p3 and p2

File: Jenkinsfile (hunk 5 above)
Comment (zxy844288792) [reply Jenkinsfile]: I have just merged the latest commit

**Comment by hcho3** [comment]
@yongwww Eventually, we should do all our tests inside a Docker container, so that we have reproducible installation and build. In particular, we can store the exact version of Tensorflow inside the Dockerfile.

**Pull request merged**